### DISCUSSION TITLE
Type of sigma in BC.

### DISCUSSION THREAD

**Discussion by nibeditabh** [ashwinktpu/StarPlat#4] [General]
For BC, with sigma as float type, I noticed it was printing out not a number (nan) for some nodes(in USRoad Network). I tried changing it to double, it worked fine. Please..check the BC output with your implementation too. Based on that we can change the sigma propNode's type in DSL code.

**Comment by mrprajesh** [comment]
In CUDA we are using this BC DSL. Two changes we did. 
1. Made `double` for variables: sigma, delta, and BC. //to avoid automatic/implicit type conversion
2. Moved the propNode decalration out of the for loop as suggested by the new team member.

Named as `bc_dsl_v3`. Let us know your thoughts or if there is problem.

```
function Compute_BC(Graph g, propNode <double> BC, SetN<g> sourceSet)
{
  g.attachNodeProperty(BC=0);

  propNode <double> sigma;
  propNode <double> delta;

  for (src in sourceSet) {
    g.attachNodeProperty(delta=0);
    g.attachNodeProperty(sigma=0);

    src.sigma = 1;

    iterateInBFS(v in g.nodes() from src) {
      for(w in g.neighbors(v)) {
        w.sigma=w.sigma+v.sigma;
      }
    }

    iterateInReverse(v!=src) {
      for(w in g.neighbors(v)) {
        v.delta = v.delta + (v.sigma / w.sigma) * ( 1 + w.delta);
      }
      v.BC = v.BC + v.delta;
    }
  }
}

```
